src/credentials/attestation_verifier/JSONFileRevocationSet.cpp

```diff
@@ -0,0 +1,189 @@
+            Json::Value jsonCrlRevokedSerialNumbers =
+                mRevocationSet[static_cast<int>(revocation_set_idx)]["revoked_serial_numbers"];
+
+            uint8_t crlIssuerSubjectKeyIdBuf[Crypto::kAuthorityKeyIdentifierLength] = { 0 };
+            ByteSpan crlIssuerSubjectKeyId(crlIssuerSubjectKeyIdBuf);
+
+            VerifyOrReturnError(Encoding::HexToBytes(jsonCrlIssuerSubjectKeyId.asString().c_str(),
+                                                     jsonCrlIssuerSubjectKeyId.asString().size(), crlIssuerSubjectKeyIdBuf,
+                                                     sizeof(crlIssuerSubjectKeyIdBuf)) == Crypto::kAuthorityKeyIdentifierLength,
+                                AttestationVerificationResult::kInvalidArgument);
+
+            size_t crlSignerCertificateMaxLength = BASE64_MAX_DECODED_LEN(jsonCrlIssuerName.asString().size());
+
+            Platform::ScopedMemoryBuffer<uint8_t> crlSignerCertificate;
+
+            ReturnErrorCodeIf(!crlSignerCertificate.Alloc(crlSignerCertificateMaxLength), AttestationVerificationResult::kNoMemory);
+
+            // 2.
+            size_t crlSignerCertificateLength =
+                Base64Decode(jsonCrlIssuerName.asString().c_str(), jsonCrlIssuerName.asString().size(), crlSignerCertificate.Get());
+            VerifyOrReturnError(crlSignerCertificateLength > 0 && crlSignerCertificateLength != UINT16_MAX,
+                                AttestationVerificationResult::kInternalError);
+
+            // 3.
+            if (isPaa)
+            {
+                Crypto::AttestationCertVidPid vid;
+
+                Crypto::ExtractVIDPIDFromAttributeString(
+                    Crypto::DNAttrType::kCommonName, ByteSpan(crlSignerCertificate.Get(), crlSignerCertificateLength), vid, vid);
+
+                if (vid.mVendorId.HasValue())
+                {
+                    if (vid.mVendorId.Value() != vidPidUnderTest.mVendorId.Value())
+                    {
+                        // VID does not match. Stop further processing and continue to next entry.
+                        continue;
+                    }
+                }
+            }
+            // 4.
+            else
+            {
+                Crypto::AttestationCertVidPid vidPid;
+
+                Crypto::ExtractVIDPIDFromAttributeString(Crypto::DNAttrType::kCommonName,
+                                                         ByteSpan(crlSignerCertificate.Get(), crlSignerCertificateLength), vidPid,
+                                                         vidPid);
+
+                if (vidPid.mVendorId.HasValue() && vidPid.mVendorId.Value() != vidPidUnderTest.mVendorId.Value())
+                {
+                    // VID does not match. Stop further processing and continue to next entry.
+                    continue;
+                }
+
+                if (vidPid.mProductId.HasValue())
```

Comment:
So this is the only thing different between the isPaa and not branches, right?  Why isn't it just this part that's conditioned on `isPaa`?  Or am I missing some other subtle difference?

Reply:
PAA does not contain Product ID (PID)

Reply:
My point is that we could have all the code be shared except:
```
  if (isPaa)
  {
      // Make sure there is no PID
  }
  else
  {
     // Check that the PID matches
  }
```
or so.

Reply:
Agreed. Fixed in latest commit